parser.d.ts

```diff
@@ -0,0 +1,159 @@
+type _If<Test, Then, Else> = Test extends true ? Then : Else;
+
+export type Features = {
+  lookbehind?: boolean;
+  namedGroups?: boolean;
+  unicodePropertyEscape?: boolean;
+  unicodeSet?: boolean;
+};
+
+export type AstNodeType =
+  | "alternative"
+  | "anchor"
+  | "characterClass"
+  | "characterClassEscape"
+  | "characterClassRange"
+  | "disjunction"
+  | "dot"
+  | "group"
+  | "quantifier"
+  | "reference"
+  | "unicodePropertyEscape"
+  | "value";
+
+export type Base<T extends AstNodeType> = {
+  range: [number, number];
+  raw: string;
+  type: T;
+};
+
+export type AstNode<F extends Features = {}> =
+  | Alternative<F>
+  | Anchor
+  | CharacterClass<F>
+  | CharacterClassEscape
+  | CharacterClassRange
+  | Disjunction<F>
+  | Dot
+  | Group<F>
+  | Quantifier<F>
+  | Reference<F>
+  | UnicodePropertyEscape
+  | Value;
+
+export type RootNode<F extends Features = {}> =
+  | Exclude<AstNode<F>, CharacterClassRange | UnicodePropertyEscape>
+  | _If<F["unicodePropertyEscape"], UnicodePropertyEscape, never>;
```

Comment:
Rather than excluding-and-reincluding `UnicodePropertyEscape` here, is it possible to just move `_If<F["unicodePropertyEscape"], UnicodePropertyEscape, never>` to the `AstNode<F>` type?

Reply:
done